api/envoy/config/filter/udp/dns_filter/v2alpha/dns_filter.proto

```diff
@@ -0,0 +1,70 @@
+syntax = "proto3";
+
+package envoy.config.filter.udp.dns_filter.v2alpha;
+
+import "envoy/api/v2/core/base.proto";
+
+import "google/protobuf/duration.proto";
+
+import "udpa/annotations/status.proto";
+
+import "udpa/annotations/migrate.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.config.filter.udp.dns_filter.v2alpha";
+option java_outer_classname = "DnsFilterProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_migrate).move_to_package =
+    "envoy.extensions.filter.udp.dns_filter.v3";
+option (udpa.annotations.file_status).work_in_progress = true;
+
+// [#protodoc-title: DNS Filter]
+// DNS Filter :ref:`configuration overview <config_udp_listener_filters_dns_filter>`.
+// [#extension: envoy.filters.udp_listener.dns_filter]
+
+// Configuration for the DNS filter.
+message DnsFilterConfig {
+  message AddressList {
+    repeated string address = 1;
+  }
+
+  // This message type is extensible and can contain a list of addresses
+  // or dictate some other method for resolving the addresses for an
+  // endpoint
+  message DnsEndpoint {
+    oneof endpoint_config {
+      AddressList addresslist = 1;
+    }
+  }
+
+  message DnsVirtualDomain {
+    // The domain name for which Envoy will respond to query requests
+    string name = 1 [(validate.rules).string = {min_len: 2 well_known_regex: HTTP_HEADER_NAME}];
+
+    // The configuration containing the method to determine the address
+    // of this endpoint
+    DnsEndpoint endpoint = 2;
+
+    // Sets the TTL in dns answers from Envoy returned to the client
+    google.protobuf.Duration answer_ttl = 3 [(validate.rules).duration = {gt {}}];
+  }
+
+  // This message contains the configuration for the Dns Filter operating
+  // in a server context. This message will contain the virtual hosts and
+  // associated addresses with which Envoy will respond to queries
+  message ServerCtxConfig {
+    uint32 retry_count = 1;
+
+    // Fully qualified domain names for which Envoy will respond to queries
+    repeated DnsVirtualDomain virtual_domains = 2 [(validate.rules).repeated = {min_items: 1}];
+
```

Comment:
Can we have a list of known domain names that the envoy dns filter is authoritative for?

This will dictate the behavior when there is no dns entry for a given domain name and if the query must be forwarded up to the external server or not.

For e.g, if the dns query for "google.com" comes in , we will know it isn't something the filter is responsible to know because "google.com" is not in the list of known domain names.
On the  contrary, lets say a query for "foo.envoycluster1.com" comes in and the known domain name has an entry for "envoycluster1.com", we check if the entry for foo exists. If not, we immediately return with NXDOMAIN. There is no point in forwarding such queries to an external dns server

Reply:
More details on what I mean here - https://github.com/sumukhs/envoy-dnsfilter/blob/master/src/dns.proto#L34

Reply:
Sure, I can add a separate list for "known domains" vs "known names"